qiskit/circuit/parameterexpression.py

```diff
@@ -306,12 +334,71 @@ def log(self):
         from sympy import log as _log
         return self._call(_log)
 
+    def sqrt(self):
+        """Return square root of ParameterExpression"""
+        from sympy import sqrt as _sqrt
+        return self._call(_sqrt)
+
+    def __floor__(self):
+        """Return floor of ParameterExpression."""
+        from sympy import floor as _floor
+        return self._call(_floor)
+
+    def __ceil__(self):
+        """Return floor of ParameterExpression."""
+        from sympy import ceiling as _ceil
+        return self._call(_ceil)
+
+    def max(self, **kwargs):
+        """Return max expression.
+
+        This is called as numpy.max(expr1, expr2).
+
+        Note: python's max(expr1, expr2) won't work.
```

Comment:
Indeed, it does not work but also does not fail. For example:
```
alpha = Parameter('α')
beta = Parameter('β')
gamma = max(alpha, beta+1)
gamma
```
prints `ParameterExpression(β + 1)` which is wrong. While
```
alpha = Parameter('α')
beta = Parameter('β')
gamma = np.max(alpha, beta+1)
gamma
```
prints `ParameterExpression(Max(α, β + 1))` which is correct. Is there some way to either

1. get python's `max` working or
2. get it to raise an error when `ParameterExpression`s are encountered?

Having `max` return an incorrect results will create errors that are very hard to trace if no warning or error is raised.

Reply:
It currently raises an error when ParameterExpressions are encountered.